AP_HAL_ChibiOS:  Add HGLRC_H743_PRO board support (#33873)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/HGLRC_H743_PRO/README.md

## Where to Buy

Available from the [HGLRC flight controller store](https://www.hglrc.com/collections/fc).

## Features

- STM32H743VI microcontroller
- ICM42688-P IMU
- SPA06-003 barometer
- AT7456E analog OSD
- 16 MiB onboard flash for data logging
- 8 motor outputs
- 7 UARTs, with UART5 exposed as RX-only
- I2C connector for external peripherals
- Addressable LED output
- Dual camera inputs with an analog camera switch
- Switchable 10V VTX power output
- USB Type-C connector
- 5V 2A and 10V 2A BECs
- Up to 12S battery input

## Board Images

![HGLRC H743 PRO top](board-top.png "HGLRC H743 PRO top")

![HGLRC H743 PRO bottom](board-bottom.png "HGLRC H743 PRO bottom")

## UART Mapping

The UART pins are marked `Rn` and `Tn` on the board. `Rn` is the receive pin and `Tn` is the transmit pin for UARTn.

| ArduPilot port | Hardware port | Default use | TX DMA | RX DMA |
|----------------|---------------|-------------|--------|--------|
| SERIAL0 | USB | MAVLink | Yes | Yes |
| SERIAL1 | USART1 | VTX (MSP DisplayPort) | Yes | Yes |
| SERIAL2 | USART2 | RC input (CRSF/ELRS) | Yes | Yes |
| SERIAL3 | USART3 | GPS or general-purpose serial | Yes | Yes |

> | SERIAL4 | UART4 | General-purpose serial | Yes | Yes |

Reviewer: ```suggestion
| SERIAL4 | UART4 (RX only) | General-purpose serial | ---| Yes |
```

| SERIAL5 | UART5 RX | ESC telemetry (RX-only) | -- | Yes |
| SERIAL6 | -- | Reserved | -- | -- |
| SERIAL7 | UART7 | GPS | Yes | Yes |
| SERIAL8 | UART8 | General-purpose serial | Yes | Yes |

## RC Input

> UART2 is configured by default for serial RC input and supports protocols such as CRSF and ELRS. Connect the receiver to the `T2` and `R2` pads. Set `SERIAL2_OPTIONS` to `0` for CRSF/ELRS.

Reviewer: ```suggestion
UART2 is configured by default for serial RC input and supports all ArduPilot RC protocols but not PPM. Connect the receiver to the `T2` and `R2` pads and set `SERIAL2_OPTIONS` appropriately.
```

> UART4 can be configured for SBUS or another serial RC protocol. PPM input is not supported.

Reviewer: remove

| GPIO(58) | PWM 9 / addressable LED |
| GPIO(80) | Buzzer |
| GPIO(81) | VTX power control |
| GPIO(82) | Camera switch |

## Battery Monitoring

The board has an onboard battery voltage sensor and an ESC current-sensor input. The default firmware settings are:

- `BATT_MONITOR=4`
- `BATT_VOLT_PIN=10`
- `BATT_CURR_PIN=11`
- `BATT_VOLT_MULT=21.0`
- `BATT_AMP_PERVLT=120`

Current monitoring should be calibrated for the connected ESC.

## Compass

The HGLRC H743 PRO does not have a built-in compass. An external compass can be connected to the `SCL` and `SDA` pads on the I2C connector.

## VTX Power Control

The 10V VTX supply is controlled by GPIO 81 and is assigned to Relay 1 by default. It is enabled at startup. Assign an RC channel the `Relay1 Control` auxiliary function to switch VTX power in flight.

## Camera Switch

The analog video input can be switched between `C1` and `C2` using GPIO 82, which is assigned to Relay 2 by default. Assign an RC channel the `Relay2 Control` auxiliary function to select the camera.

## Firmware

Firmware for the HGLRC H743 PRO can be found [here](https://firmware.ardupilot.org) in sub-folders labeled `HGLRC_H743_PRO`.

## Loading Firmware

> To load ArduPilot firmware, hold the boot button or pad low while powering up the board. Then use STM32CubeProgrammer or another DFU utility to load the `HGLRC_H743_PRO_bl.bin` file. Subsequently, a ground control station can update firmware using the `*.apj` files.

Reviewer: Name the button by its actual silkscreen label so a user can find it — it's the switch next to the USB-C connector, labeled `PWR BUT` on `board-bottom.png`.
```suggestion
To load ArduPilot firmware, hold the `PWR BUT` button (next to the USB-C connector) low while powering up the board. Then use STM32CubeProgrammer or another DFU utility to load the `HGLRC_H743_PRO_bl.bin` file. Subsequently, a ground control station can update firmware using the `*.apj` files.
```
